jobs/secureproxy/templates/config/nginx.conf.erb

```diff
@@ -179,7 +179,12 @@ http {
       add_header Strict-Transport-Security $sts always;
       add_header X-Content-Type-Options $content_type_options always;
       add_header X-XSS-Protection $xss_protection always;
-      add_header Content-Type $default_content_type always;
+
+      # Don't add a `Content-Type` header if the response length is 0,
+      # e.g. 204/304 status codes
+      if ( $upstream_response_length != 0 ) {
```

Comment:
this is making me think of all the `if is evil` posts about nginx. I think using a `map` is safer here (but I'm still pondering how exactly to write it)

Reply:
Yeah, I couldn't find the original "if is evil" post for context. All I found was people saying not to use `if` in `location` blocks

Reply:
yeah, the `not in location blocks` thing is what I mostly found while digging. I think this is the original "If is evil" post: https://github.com/nginxinc/nginx-wiki/blob/master/source/start/topics/depth/ifisevil.rst


(but note that we are in a location block)

Reply:
Here's the original post: https://web.archive.org/web/20150924145945/https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/

It seems like it is saying not to use `if` inside a location block, but we're using it inside a `server` block.

All that being said, if there's a safer way to do it, I'm all for it. It seems like the only way to do it would probably be some kind of Lua block, but that felt beyond me

Reply:
@bengerman13 What do you think about 099ce0115a72c958e457ef852eb78f5dd8898ab1 ?

Inspired by: 

https://stackoverflow.com/questions/33660493/nginx-lua-set-proxy-headers
https://www.reddit.com/r/nginx/comments/4clta3/using_lua_to_set_headers/

I could also use this to set the headers: https://github.com/openresty/headers-more-nginx-module?tab=readme-ov-file#more_clear_headers, which even allows you to specifically enumerate the response codes for which the header should be added. But it seems to me like we would rather enumerate response codes where the header **should not** be added. And basing the condition on the response length is probably a better way regardless.